tests/conftest.py

```diff
@@ -220,7 +221,21 @@ def application_process(
 
 @fixture(scope="session")
 def docker_client() -> DockerClient:
-    return docker.from_env()
+    tests_id = secrets.token_hex(5)
+    docker_client = docker.from_env()
+    setattr(docker_client, "test_id", tests_id)
```

Comment:
```suggestion
    setattr(docker_client, "_gprofiler_test_id", tests_id)
```

As you're patching a global object - refrain from patching possibly used attributes. The `_gprofiler_` prefix ensures it's not used.